File: app-java/src/main/java/science/monke/spring/config/CustomProperties.java
package science.monke.spring.config;

import lombok.Getter;
import lombok.Setter;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.context.annotation.Configuration;

@Getter
@Setter
@Configuration
@ConfigurationProperties(prefix = "custom")
public class CustomProperties {
  private Region region;
  private Environment environment;
}

File: app-kotlin/src/main/kotlin/science/monke/spring/config/CustomProperties.kt
package science.monke.spring.config;

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.boot.context.properties.ConstructorBinding

@ConstructorBinding
@ConfigurationProperties(prefix = "custom")
data class CustomProperties(
    val region: Region,
    val environment: Environment
)
